integration/distinct_test.go

```diff
@@ -121,3 +123,66 @@ func TestDistinctErrors(t *testing.T) {
 		})
 	}
 }
+
+func TestDistinctDuplicates(t *testing.T) {
+	t.Parallel()
+
+	ctx, coll := setup.Setup(t)
+
+	for name, tc := range map[string]struct {
+		docs     []bson.D
+		key      string
+		expected []any
+	}{
+		"LongDouble": {
+			docs: []bson.D{
+				{{"v", int64(42)}},
+				{{"v", float64(42)}},
+				{{"v", "42"}},
+			},
+			key:      "v",
+			expected: []any{int64(42), "42"},
+		},
+		"LongInt": {
+			docs: []bson.D{
+				{{"v", int64(42)}},
+				{{"v", float64(42)}},
+				{{"v", "42"}},
+			},
+			key:      "v",
+			expected: []any{int64(42), "42"},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			var docs = make([]any, len(tc.docs))
+
+			for i, doc := range tc.docs {
+				docs[i] = doc
+			}
+
+			_, err := coll.InsertMany(ctx, docs)
+			require.NoError(t, err)
+
+			distinct, err := coll.Distinct(ctx, tc.key, bson.D{})
+			require.NoError(t, err)
+
+			assert.Equal(t, len(tc.expected), len(distinct), fmt.Sprint(distinct))
+
+			for i, value := range distinct {
+				expectedValue := tc.expected[i]
+
+				if value != expectedValue {
+					// Values order is random so we cannot expect the types to be the same
```

Comment:
> Does this mean that sometimes the test may return []{"42", int64(42)} instead of []{int64(42), "42"}?

Yes

> If that's the case, could we check slice.Contains for possible expected types. FYI, it would be types in docs because expected only contains subset of the types.

What's the reason behind using `slice.Contains` here? Currently we iterate through array, compare types and if they're not the same we compare values if they have specific types.

Reply:
Let's say we have `expected: []any{int64(42), "42"}` in `LongDouble` test case, imagine if distinct returns `[]{"42", int64(42)}` (this is possible right?), for 0-th index we have expected `int64(42)` and res from distinct string `"42"` right? I don't know how that is working, because they are not equal :thinking: I think I'm misunderstanding something here :see_no_evil: 

Hence I suggested using slice.Contains to check containment instead of equality.